src/robothub_oak/live_view.py

```diff
@@ -0,0 +1,290 @@
+
+        encoder = device.pipeline.createVideoEncoder()
+        encoder_profile = dai.VideoEncoderProperties.Profile.H264_MAIN
+        encoder.setDefaultProfilePreset(fps, encoder_profile)
+        encoder.input.setQueueSize(1)
+        encoder.input.setBlocking(False)
+        encoder.setKeyframeFrequency(int(fps))
+        encoder.setBitrate(1500 * 1000)
+        encoder.setRateControlMode(dai.VideoEncoderProperties.RateControlMode.CBR)
+        encoder.setNumFramesPool(3)
+
+        component.node.video.link(encoder.input)
+
+        def encoded(pipeline, device):
+            xout = XoutH26x(
+                frames=StreamXout(encoder.id, encoder.bitstream),
+                color=True,
+                profile=encoder_profile,
+                fps=encoder.getFrameRate(),
+                frame_shape=component.node.getResolution()
+            )
+            xout.name = component._source
+            return component._create_xout(pipeline, xout)
+
+        return encoded
+
+    @staticmethod
+    def is_h264(component: Component) -> bool:
+        encoder = None
+        if isinstance(component, CameraComponent) or isinstance(component, StereoComponent):
+            encoder = component.encoder
+        elif isinstance(component, NNComponent):
+            encoder = component._input.encoder
+
+        if not encoder:
+            raise ValueError('This component does not have an encoder.')
+
+        if encoder.getProfile() != dai.VideoEncoderProperties.Profile.H264_MAIN:
+            return False
+
+        return True
+
+    @staticmethod
+    def get_stream_size(component) -> Tuple[int, int]:
+        if isinstance(component, CameraComponent):
+            return component.stream_size
+        elif isinstance(component, StereoComponent):
+            return component.left.stream_size
+        elif isinstance(component, NNComponent):
+            return component._input.stream_size
+
+    @staticmethod
+    def get(unique_key: str = None, title: str = None):
+        if title is not None:
+            return LiveView.get_by_name(title)
+        elif unique_key is not None:
+            return LiveView.get_by_unique_key(unique_key)
+        else:
+            raise ValueError('Either name or unique_key must be specified.')
+
+    @staticmethod
+    def get_by_title(title: str) -> Optional['LiveView']:
+        for live_view in LIVE_VIEWS.values():
+            if live_view.title == title:
+                return live_view
+        return None
+
+    @staticmethod
+    def get_by_unique_key(unique_key: str) -> Optional['LiveView']:
+        if unique_key not in LIVE_VIEWS:
+            raise ValueError(f'Live View with unique_key {unique_key} does not exist.')
+
+        return LIVE_VIEWS[unique_key]
+
+    def add_rectangle(self, rectangle: BoundingBox, label: str) -> None:
+        self.rectangles.append(rectangle)
+        self.labels.append(label)
+
+    def add_bbox(self, bbox: BoundingBox, label: str) -> None:
```

Comment:
i would keep only `draw_rectangle`. Bounding box may be not understandable for the user.

Reply:
agree, what do you think about `add_rect`? also I think add suits better since we don't actually draw anything on live view, but instead pass as objects along with encoded stream

Reply:
yes, I agree, that is a good point.